src/pages/products/Products.jsx

```diff
@@ -1,27 +1,37 @@
 import { useLocation } from "react-router-dom";
-import { useEffect } from "react";
-import { Card, FilterSidebar } from "../../components";
+import { Card, FilterSidebar, Loader } from "../../components";
 import { useDocTitle } from "../../hooks/useDocTitle";
 import "./products.css";
 import { useProducts } from "../../context";
+import { useState, useEffect } from "react";
 
 export function Products() {
   useDocTitle("Products - SmashCart - Manoj Sarna");
+  const [loading, setLoading] = useState();
+  const { products, finalFilteredData } = useProducts();
+  useEffect(() => {
+    setLoading(true);
+    const timeout = setTimeout(() => {
+      setLoading(false);
+    }, 700);
+    return () => clearTimeout(timeout);
+  }, [products]);
+
   const { pathname } = useLocation();
   useEffect(() => {
-    window.scrollTo(0, 0);
+    window.scrollTo({ behavior: "smooth", top: "0px" });
   }, [pathname]);
 
-  const { products, finalFilteredData } = useProducts();
-
-  return (
+  return loading ? (
+    <Loader show={loading} />
+  ) : (
     <main className="sm-main">
       <div className="sm-main-product-page">
         <FilterSidebar />
 
         <div className="sm-main-products-container">
           {finalFilteredData.map((product) => (
-            <Card key={product._id} product={product} />
+            <Card key={product._id} product={product} btnName={"Add To Cart"} />
```

Comment:
Appreciation: you have made the card component more modular by adding button fuctionality.